public/js/application.js

```diff
@@ -84,12 +84,16 @@ var View = {
 	},
 
 
-	updateAvailableDates: function(notAllowedDates){
-		var array = ["2013-03-14","2013-03-15","2013-03-16"]
+	updateAvailableDates: function(departTrips, returnTrips){
+		// var array = ["2013-03-14","2013-03-15","2013-03-16"]
+		var departDates = departTrips.map(function (trip) {return trip.depart_date})
+		var returnDates = returnTrips.map(function (trip) {return trip.depart_date})
```

Comment:
Fine to do client side assuming you're passing all the trips back to the client. And best just to pass all the trips to client 

Sent from an iPhone 

> On Apr 4, 2016, at 3:34 PM, Peter Cusack <notifications@github.com> wrote:
> 
> In public/js/application.js:
> 
> > @@ -84,12 +84,16 @@ var View = {
> >  	},
> >  
> >  
> > -	updateAvailableDates: function(notAllowedDates){
> > -		var array = ["2013-03-14","2013-03-15","2013-03-16"]
> > +	updateAvailableDates: function(departTrips, returnTrips){
> > +		// var array = ["2013-03-14","2013-03-15","2013-03-16"]
> > +		var departDates = departTrips.map(function (trip) {return trip.depart_date})
> > +		var returnDates = returnTrips.map(function (trip) {return trip.depart_date})
> Should this be done here or on the server?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>